File: Java_20Q/Q12.java
public class Q12 {

  static int calls = 0;

  public static int f(int n) {
    int f;

    if (n < 2) {
      calls++;
      System.out.println(calls);
      return n;
    }
    calls++;
    System.out.println(calls);
    return f(n - 1) + f(n - 2) + f(n - 3);

  }

  public static void main(String[] args) {
    System.out.println(f(4));
  }
}

File: C++_20Q/Q12.cpp
#include <iostream>

int calls = 0;

int f(int n) {
  if (n < 2) {
    calls += 1;
    return n;
  }
  calls += 1;
  return f(n - 1) + f(n - 2) + f(n - 3);
}

int main() {
  std::cout << "Sum: " << f(4) << std::endl;
  std::cout << "Calls: " << calls << std::endl;
}
